cpp/ruapu.hpp

```diff
@@ -0,0 +1,33 @@
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#define RUAPU_IMPLEMENTATION
+#include "../ruapu.h"
+
+class ruapu
+{
+public:
+    ruapu() { ruapu_init(); }
```

Comment:
Not sure if a C++ "binding" is necessary providing that ruapu is already in C and we can call C directly in C++, but even if it does, we probably should use singleton design pattern so that `ruapu_init` only gets called once. Or at least we can do the following

```suggestion
    ruapu() { 
      static int initialised = 0;
      if (!initialised) {
        ruapu_init();
        initialised = 1;
      }
    }
```

And all methods should really be static class methods instead of instance ones because there's no underlying ruapu context/handle associated to this C++ class.